### ISSUE TITLE
关于百度网盘网页版不能调用助手的问题

### ISSUE THREAD

**Issue by willxing8** [syhyz1990/baiduyun#373]
目前发现几个问题：
1.百度网盘网页版调用下载助手均不能正常使用。出现提示：调用失败，请重新刷新页面！
2.初步怀疑是浏览器问题：用chrome 版本 113.0.5672.127（正式版本） （64 位）可以正常调用下载助手。但使用高版本不能使用助手下载，如chrome 版本 126.0.6478.115（正式版本） （64 位）,同样125版本也试过，不能用，不知道是不是兼容问题。

请大佬抽空解决或优化。

**Comment by ahdung** [comment]
我也是前两天突然发现百度网盘出不来下载助手了

**Comment by dfvips** [comment]
一样，用不了了

**Comment by Tianyi16** [comment]
我的其他网盘也没有按钮了，自从说打开开发者模式后整个脚本感觉都失效了。大佬快看看吧，没有它好不习惯啊

**Comment by Tianyi16** [comment]
我的是在所有网盘上也没有按钮，不能使用了， 自从更新打开开发者模式后整个脚本都失效了。 大神看看吧，没有它好不习惯啊

**Comment by Gentran** [comment]
```javascript
        async initPanLinker() {
            pt = this.detectPage();
            let res = await base.post
            (`https://api.youxiaohou.com/config/v2?ver=${version}&a=${author}`, {}, {}, 'text');
            pan = JSON.parse(base.d(res));
>           pan.btn.main = '.wp-s-agile-tool-bar__header'
            Object.freeze && Object.freeze(pan);
            this.addButton()
        }
    };

    let ali = {
```
Add a line like above will make the button appear in baidu netdisk

**Comment by tangkunyin** [comment]
开发模式打开，多刷几次能出来。但是下载不了，比如cURL点了之后提示：获取下载链接失败！请刷新网页后重试！

但无论怎么刷都不好使，百度这货也是真的服。。。

**Comment by dfvips** [comment]
大佬修复一下百度网盘吧
![image](https://github.com/syhyz1990/baiduyun/assets/21048630/9516917f-44eb-4d81-a01a-f589561d1e51)

**Comment by Dragon1573** [comment]
> Add a line like above will make the button appear in baidu netdisk

Added but still unable to show the button. 😞

**Comment by Gentran** [comment]
Baidu changed access token [implicit grant](https://pan.baidu.com/union/doc/6l0ryrjzv#%E7%AE%80%E5%8C%96%E6%A8%A1%E5%BC%8F) returning method recently. The access token will be return in final url like this: `https://openapi.baidu.com/oauth/2.0/login_success#expires_in=2592000&access_token=xxxxxx`. And any string follwing the # symbol in the URL will lost. This method is not suitable for the script any more.

**Comment by Dragon1573** [comment]
> And any string follwing the `＃` symbol in the URL will lost. This method is not suitable for the script any more.

Why is that? The URL response from _Baidu OAuth Server_ is a callback URL for developer's client. Client parse the URL and extract access token from it.

Maybe parser should be upgraded instead of just switching to another method?

**Comment by Gentran** [comment]
> Maybe parser should be upgraded instead of just switching to another method?

There already exist a function `base.getFinalUrl` in this script that can exact the callback URL but unfortunately as I described above the access token after `#` can not be obtained.

Actually Baidu will return a 302 redirection after a [implicit grant](https://pan.baidu.com/union/doc/6l0ryrjzv) request. But with pure javascript the `Location` header is not visible to the developer.

I already tried using `Fetch API` and `XMLHttpRequest` to get the final url and redirection `Location` header but both didn't work.

Maybe the [Device Code](https://pan.baidu.com/union/doc/fl1x114ti) mode and [Authorization Code](https://pan.baidu.com/union/doc/al0rwqzzl) mode will work but developer must update the token grant URL to add AppKey. And add addition code to display grant page and store the refresh_token.
